File: 操作系统/Windows API/Windows 程序设计/第6版/CSharp/Chapter05/ColorScrollWithValueConverter/ColorScrollWithValueConverter/DoubleToStringHexByteConverter.cs
using System;
using Windows.UI.Xaml.Data;

namespace ColorScrollWithValueConverter
{
    public class DoubleToStringHexByteConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return ((int)(double)value).ToString("X2");
        }
        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return value;
        }
    }
}

File: 操作系统/Windows API/Windows 程序设计/第6版/CPlusPlus/Chapter05/ColorScrollWithValueConverter/ColorScrollWithValueConverter/DoubleToStringHexByteConverter.cpp
#include "pch.h"
#include "DoubleToStringHexByteConverter.h"

using namespace ColorScrollWithValueConverter;

using namespace Platform;
using namespace Windows::Foundation;
using namespace Windows::UI::Xaml;
using namespace Windows::UI::Xaml::Interop;

Object^ DoubleToStringHexByteConverter::Convert(Object^ value, TypeName targetType, Object^ parameter, String^)
{
    wchar_t result[4];
    swprintf(result, 4, L"%02X", (int)(double)value);
    return ref new String(result);
}

Object^ DoubleToStringHexByteConverter::ConvertBack(Object^ value, TypeName targetType, Object^ parameter, String^)
{
    return value;
}
